java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -91,4 +89,119 @@ protected static ConnectionManager buildConnectionManager(ChannelHandler channel
     protected static CommandManager buildCommandManager(ChannelHandler channelHandler) {
         return new CommandManager(channelHandler);
     }
+
+    /**
+     * Extracts the response from the Protobuf response and either throws an exception or returns the
+     * appropriate response as an <code>Object</code>.
+     *
+     * @param response Redis protobuf message
+     * @return Response <code>Object</code>
+     */
+    protected Object handleObjectResponse(Response response) {
+        // convert protobuf response into Object
+        return new BaseCommandResponseResolver(RedisValueResolver::valueFromPointer).apply(response);
+    }
+
+    /**
+     * Checks that the Response contains the simple-string Ok.
+     *
+     * @return An Ok response
+     * @throws RedisException if there's a type mismatch
+     */
+    protected Ok handleOkResponse(Response response) {
+        Object value = handleObjectResponse(response);
+        if (value == Ok.INSTANCE) {
+            return Ok.INSTANCE;
+        }
+        String className = (value == null) ? "null" : value.getClass().getSimpleName();
+        throw new RedisException(
+                "Unexpected return type from Redis: got " + className + " expected simple-string Ok");
+    }
+
+    /**
+     * Extracts the response value from the Redis response and either throws an exception or returns
+     * the value as a <code>String</code>.
+     *
+     * @param response Redis protobuf message
+     * @return Response as a <code>String</code>
+     * @throws RedisException if there's a type mismatch
+     */
+    protected String handleStringResponse(Response response) {
+        Object value = handleObjectResponse(response);
+        if (value instanceof String || value == null) {
+            return (String) value;
+        }
+        if (value instanceof Ok) {
+            return Ok.INSTANCE.toString();
```

Comment:
that's an issue - now if the user do client.set("foo", "bar") they should expect Ok response, but if they do client.set("foo", "bar", SetOptions("NX")) and the set succeed it would return a string "Ok" response.

Reply:
If there isn't an option in java to return Ok or String, we might need to have a separate API for setGet command. lets discuss it tomorrow

Reply:
@shachlanAmazon

Reply:
I guess this condition is never true in real world (outside testing) and the example above goes thru `value instanceof String`.